core/src/main/java/org/geysermc/geyser/level/block/BlockStateValues.java

```diff
@@ -99,6 +99,25 @@ public static boolean canPistonMoveBlock(BlockState state, boolean isPushing) {
         };
     }
 
+    /**
+     * Get the type of fluid from the block state, including waterlogged blocks.
+     *
+     * @param state BlockState of the block
+     * @return The type of fluid
+     */
+    public static Fluid getFluid(int state) {
+        BlockState blockState = BlockState.of(state);
```

Comment:
BlockState could be null if the state is unknown, see e.g. https://github.com/GeyserMC/Geyser/pull/4764 - that case would throw an NPE below